src/proto/single/mod.rs

```diff
@@ -1,77 +1,122 @@
 use chrono::{DateTime, Utc};
+use derive_builder::Builder;
 use std::collections::HashMap;
 use std::io::prelude::*;
 
 mod cmd;
 mod resp;
+mod utils;
 
-use crate::error::Error;
+use crate::error::{self, Error};
 
 pub use self::cmd::*;
 pub use self::resp::*;
 
+const JOB_DEFAULT_QUEUE: &str = "default";
+const JOB_DEFAULT_RESERVED_FOR_SECS: usize = 600;
+const JOB_DEFAULT_RETRY_COUNT: usize = 25;
+const JOB_PRIORITY_MAX: u8 = 9;
+const JOB_DEFAULT_PRIORITY: u8 = 5;
+const JOB_DEFAULT_BACKTRACE: usize = 0;
+
 /// A Faktory job.
 ///
+/// To create a job, use 'Job::new' specifying 'kind' and 'args':
+/// ```
+/// use faktory::Job;
+///
+/// let _job = Job::new("order", vec!["ISBN-13:9781718501850"]);
+/// ```
+///
+/// Alternatively, use 'JobBuilder' to construct a job:
+/// ```
+/// use faktory::JobBuilder;
+///
+/// let result = JobBuilder::default()
+///     .kind("order")
+///     .args(vec!["ISBN-13:9781718501850"])
+///     .build();
+/// if result.is_err() {
+///     todo!("Handle me gracefully, please.")
+/// };
+/// let _job = result.unwrap();
```

Comment:
```suggestion
/// let _job = JobBuilder::default()
///     .kind("order")
///     .args(vec!["ISBN-13:9781718501850"])
///     .build()?;
/// # Ok(())
```

Reply:
Added usage examples taking into account that the build is not returning result any longer